libdrgn/program.c

```diff
@@ -788,20 +953,166 @@ struct drgn_error *drgn_program_find_prstatus_by_tid(struct drgn_program *prog,
 						     struct nstring *ret)
 {
 	struct drgn_error *err;
-	struct drgn_prstatus_map_iterator it;
-
-	assert(!(prog->flags & DRGN_PROGRAM_IS_LINUX_KERNEL));
```

Comment:
I think we might still want this assertion. For kernel core dumps, `drgn_program_find_thread()` doesn't return a thread with a `prstatus`. Kernel core dumps use the CPU-indexed list of `prstatus` notes instead.